lib/student.rb

```diff
@@ -0,0 +1,20 @@
+class Student
+  attr_reader :name, :age, :scores
+  require pry
+
+  def initialize(params={})
+    @name = params[:name]
+    @age = params[:age]
+    @scores = []
+  end
+
+  def log_score(score)
+    @scores << score
+  end
+
+  def grade
+    return 0 if @scores.empty?
+    total_score = @scores.reduce(:+)
```

Comment:
Another option here is @scores.sum

reduce is a great enumerable but a bit more complex